File: Coding Problems - Python/Graphs/1557 Minimum Number of Vertices to Reach All Nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        Indegree = [0] * n
        res = []
        for u, v in edges:
            Indegree[v] += 1
        
        for i in range(len(Indegree)):
            if Indegree[i] == 0:
                res.append(i)
        return res

File: Coding_Problems_Java/1557. Minimum Number of Vertices to Reach All Nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        int[] Indegree = new int[n];
        ArrayList<Integer> res = new ArrayList<>();

        for(List<Integer> i: edges){
            int u = i.get(0);
            int v = i.get(1);
            Indegree[v] += 1;
        }
        for (int i = 0;i < n; i++){
            if (Indegree[i] == 0){
                res.add(i);
            }
        }
        return res;
    }
}
